ci/timeout_with_stack.py

```
@@ -0,0 +1,326 @@
+    capture_all_stacks : For capturing stack traces of processes.
+
+    Examples
+    --------
+    >>> from timeout_with_stack import run_with_timeout
+    >>> exit_code = run_with_timeout(["sleep", "10"], timeout=5, enable_python=True)
+    >>> print(f"Process exited with code: {exit_code}")
+    """
+    # Start the process with a new process group
+    # Note: preexec_fn is used here as we need to create a new process group
+    # for proper termination of child processes
+    process = subprocess.Popen(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)